src/AutoRest.CSharp.V3/ClientModels/ModelBuilder.cs

```diff
@@ -13,18 +13,26 @@ internal class ModelBuilder
         private static ClientModel BuildClientEnum(SealedChoiceSchema sealedChoiceSchema) => new ClientEnum(
             sealedChoiceSchema,
             sealedChoiceSchema.CSharpName(),
-            sealedChoiceSchema.Choices.Select(c => new ClientEnumValue(ModelExtensions.CSharpName((ChoiceValue) c), ClientModelBuilderHelpers.StringConstant(c.Value))));
+            sealedChoiceSchema.Choices.Select(c => new ClientEnumValue(c.CSharpName(), ClientModelBuilderHelpers.StringConstant(c.Value))));
 
         private static ClientModel BuildClientEnum(ChoiceSchema choiceSchema) => new ClientEnum(
             choiceSchema,
             choiceSchema.CSharpName(),
             choiceSchema.Choices.Select(c => new ClientEnumValue(c.CSharpName(), ClientModelBuilderHelpers.StringConstant(c.Value))),
             true);
 
-        private static ClientModel BuildClientObject(ObjectSchema objectSchema) => new ClientObject(
-            objectSchema, objectSchema.CSharpName(),
-            objectSchema.Properties.Where(property => !(property.Schema is ConstantSchema)).Select(CreateProperty),
-            objectSchema.Properties.Where(property => property.Schema is ConstantSchema).Select(CreateConstant));
+        private static ClientModel BuildClientObject(ObjectSchema objectSchema)
+        {
+            var inheritsFrom = objectSchema.Parents?.Immediate.OfType<ObjectSchema>().SingleOrDefault();
```

Comment:
Keep in mind that we'll have to support inheritance from schemas of types other than ObjectSchema (such as OrSchema, XorSchema, NotSchema(?), GroupSchema(?)), and DictionarySchema) later on. You can get a good idea of what may/may not apply looking at the `Schemas` class in CodeModel.cs or looking at types driving from `ComplexSchema` in there too.